consensus/hotstuff/verification/combined_verifier_v2.go

```diff
@@ -141,6 +141,18 @@ func (c *CombinedVerifier) VerifyQC(signers flow.IdentityList, sigData []byte, b
 	// TODO: update to use module/signature.PublicKeyAggregator
 	aggregatedKey, err := crypto.AggregateBLSPublicKeys(signers.PublicStakingKeys()) // caution: requires non-empty slice of keys!
 	if err != nil {
+		// `AggregateBLSPublicKeys` returns a `crypto.invalidInputsError` in two distinct cases:
+		//  (i) In case no keys are provided, i.e.  `len(signers) == 0`.
+		//      This scenario _is expected_ during normal operations, because a byzantine
+		//      proposer might construct an (invalid) QC with an empty list of signers.
+		// (ii) In case some provided public keys are invalid.
+		//      This scenario is _not expected_ during normal operations, because all keys are
+		//      supposed to be checked _before_ they enter the protocol state bases on a proof
+		//      of possession. BUT, we don't have proof of possessions implemented at the moment.
```

Comment:
This not quite correct. PoP is not related to the possible error here. 

I'll try to explain:
The API of `AggregateBLSPublicKeys` accepts `PublicKey` elements and each has a signature algorithm attached to it. The function errors only if the algorithm is not `BLS_BLS12-381` (that's BLS signature on a curve called BLS12-381). There is no any other cryptographic check done. The public APIs available only allow generating valid BLS `PublicKey` structs and there is no way to play with the inner algorithm information. The error on this function can only happen if a developer mistakenly used let's say an `ECDSA` key. 

Now I wanted to clarify how the staking keys are validated. The staking contract receives bytes and validate that they represent a BLS key for the BLS12-381 group (a specific subgroup of the entire curve). The validation steps are:
 1. the bytes need to deserialize to a point on the elliptic curve BLS12-381 
 2. the point obtained is on the correct subgroup of BLS12-381 
 3. The PoP is valid
 Today, the contract only implements 1 and 2. Once the 2 steps are valid, the node software generates a `PublicKey` struct to be used by the rest of the protocol software (including hotstuff). The key struct has the correct assigned algorithm. In particular, it passes the check of `AggregateBLSPublicKeys`.
 The non-implemented step 3 of the validation does not impact this process at all. 
 
 Why is step 3 needed? Once we verify aggregated signatures using aggregated keys, we do need an extra measure to protect against Rogue attacks. Many measures are available and PoP is one of them and the most convenient for our case.

Reply:
For the comments above, we should not mention PoP, I suggest:
```
//      This scenario is _not expected_ during normal operations, because all staking keys are guaranteed by the protocol to be BLS keys.
```

Reply:
Updated the comments here : https://github.com/onflow/flow-go/pull/1831/commits/865febd177a21af2e5af1dc551f2d02ac92f75f5